src/run_tests.py

```diff
@@ -0,0 +1,52 @@
+import os
+import argparse
+import subprocess
+from colorama import Fore, Style
+
+
+def extract_files_with_extension(dir_name, extension):
+    return [name for name in os.listdir(dir_name) if name.endswith(extension)]
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Run the integration tests.')
+    parser.add_argument('-d', '--test-dir', type=str,
+                        help='The test directory.', required=True)
+    args = parser.parse_args()
+    subprocess.run('cd {:} && make clean && make'.format(args.test_dir),
+                   shell=True, stdout=subprocess.PIPE)
+    c_files = extract_files_with_extension(args.test_dir, '.c')
+    bin_names = [name[:name.rfind('.')] for name in c_files]
```

Comment:
You could simplify this with `os.path.splitext`:
https://docs.python.org/2/library/os.path.html#os.path.splitext

Reply:
Good point!